plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java

```diff
@@ -243,6 +253,676 @@ public void testFlushPartitionCache()
         computeActual(format("DROP TABLE %s", fullyQualifiedTestTableName));
     }
 
+    @Test
+    public void testPartitionProjectionOnVarCharColumnWithEnumProjectionAndStorageLocationTemplate()
+    {
+        String tableName = "nation_" + randomTableSuffix();
+        String fullyQualifiedTestTableName = getTestTableName(tableName);
+
+        computeActual(getCreateTableStatement(
+                fullyQualifiedTestTableName,
+                ImmutableList.of(
+                        "short_name1 varchar(152)",
+                        "short_name2 varchar(152)"),
+                ImmutableList.of(
+                        "partitioned_by=ARRAY['short_name1', 'short_name2']",
+                        "\"projection.enabled\"=true",
+                        "\"projection.short_name1.type\"='enum'",
+                        "\"projection.short_name1.values\"=ARRAY['PL1', 'CZ1']",
+                        "\"projection.short_name2.type\"='enum'",
+                        "\"projection.short_name2.values\"=ARRAY['PL2', 'CZ2']",
+                        "\"storage.location.template\"=" + format(
+                                "'s3a://%s/%s/%s/short_name1=${short_name1}/short_name2=${short_name2}/'",
+                                this.bucketName, HIVE_TEST_SCHEMA, tableName))));
+        computeActual(createInsertStatement(
+                fullyQualifiedTestTableName,
+                ImmutableList.of(
+                        ImmutableList.of("'POLAND_1'", "'Comment'", "0", "5", "'PL1'", "'PL2'"),
+                        ImmutableList.of("'POLAND_2'", "'Comment'", "1", "5", "'PL1'", "'CZ2'"),
+                        ImmutableList.of("'CZECH_1'", "'Comment'", "2", "5", "'CZ1'", "'PL2'"),
+                        ImmutableList.of("'CZECH_2'", "'Comment'", "3", "5", "'CZ1'", "'CZ2'"))));
+
+        assertQuery(
+                format("SELECT name FROM %s WHERE short_name1='PL1' AND short_name2='CZ2'", fullyQualifiedTestTableName),
+                "VALUES 'POLAND_2'");
+
+        assertQuery(
+                format("SELECT name FROM %s WHERE short_name1='PL1'", fullyQualifiedTestTableName),
+                "VALUES ('POLAND_1'), ('POLAND_2')");
+
+        assertQuery(
+                format("SELECT name FROM %s", fullyQualifiedTestTableName),
+                "VALUES ('POLAND_1'), ('POLAND_2'), ('CZECH_1'), ('CZECH_2')");
+    }
+
+    @Test
+    public void testPartitionProjectionOnVarCharColumnWithEnumProjection()
+    {
+        String tableName = "nation_" + randomTableSuffix();
+        String fullyQualifiedTestTableName = getTestTableName(tableName);
+
```

Comment:
There is a repetitive pattern in these tests.
Can you extract a generic method to get to a more concise form for the tests?

Reply:
Applied